### ISSUE TITLE
Headline with code block not rendered correctly sometimes

### ISSUE THREAD

**Issue by xoxys** [gohugoio/hugo#7173]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
Hugo Static Site Generator v0.68.1 linux/amd64 BuildDate: 2020-03-23T11:18:32Z
</pre>

I have some headlines which looks like this:
```
## `records(self, zone, params={})`
```
While this is rendered fine with `blackfriday`,
![grafik](https://user-images.githubusercontent.com/3391958/79306432-7d26bf00-7ef5-11ea-8720-bba24498c0ea.png)

with `goldmark` it is broken
![grafik](https://user-images.githubusercontent.com/3391958/79306590-c7a83b80-7ef5-11ea-8958-db15435acd6c.png)

There seems to be something special with the `{}` in combination with code blocks + headline.

```
## `records(self, zone, params={})`   -> broken
`records(self, zone, params={})`        -> works
## `records(self, zone, params={ })`   -> works
```

**Comment by bep** [comment]
Thanks for the report. This looks like it could be an issue with Goldmark. Could you create an issue at https://github.com/yuin/goldmark/issues and link back to this issue so we could track it?

**Comment by xoxys** [comment]
Of course, I was just not sure where to start :)
https://github.com/yuin/goldmark/issues/123

**Comment by bep** [comment]
I suspect this is the attribute parsing (used to set manual IDs on titles).

**Comment by moorereason** [comment]
This has been fixed since v0.72 when we updated to goldmark v1.31.1.
